applications/tasks.py

```diff
@@ -8,53 +7,120 @@
 from applications.enums import ApplicantTypeChoice, Weekday
 from applications.models import Address, AllocatedTimeSlot, Organisation, Person
+from common.date_utils import local_end_of_day, local_start_of_day
 from common.utils import translate_for_user
-from opening_hours.errors import ReservableTimeSpanClientError
-from opening_hours.utils.reservable_time_span_client import ReservableTimeSpanClient
+from opening_hours.enums import HaukiResourceState
+from opening_hours.utils.hauki_api_client import HaukiAPIClient
+from opening_hours.utils.time_span_element import TimeSpanElement
 from reservations.enums import CustomerTypeChoice, ReservationStateChoice, ReservationTypeChoice
 from reservations.models import RecurringReservation
 from reservations.tasks import create_or_update_reservation_statistics, update_affecting_time_spans_task
 from tilavarauspalvelu.celery import app
 from utils.sentry import SentryLogger
 
-if TYPE_CHECKING:
-    from opening_hours.utils.time_span_element import TimeSpanElement
+
+def _get_series_override_closed_time_spans(allocations: list[AllocatedTimeSlot]) -> dict[int, list["TimeSpanElement"]]:
+    """
+    Find all closed opening hours for all reservation units where allocations were made.
+    Check against these and not fully normalized opening hours since allocations can be made
+    outside opening hours (as this system defines them), but should not be on explicitly
+    closed hours, like holidays.
+    """
+    closed_time_spans: dict[int, list[TimeSpanElement]] = {}
+    for allocation in allocations:
+        resource = allocation.reservation_unit_option.reservation_unit.origin_hauki_resource
+        if resource is None or resource.id in closed_time_spans:
+            continue  # Skip if already fetched
+
+        application_round = allocation.reservation_unit_option.application_section.application.application_round
+
+        # Fetch periods from Hauki API
+        date_periods = HaukiAPIClient.get_date_periods(
+            hauki_resource_id=resource.id,
+            start_date_lte=application_round.reservation_period_end.isoformat(),  # Starts before period begins
+            end_date_lte=application_round.reservation_period_begin.isoformat(),  # Ends after period begins
+            end_date_gte="-1d",  # End date must be after yesterday
+        )
+
+        # Convert periods to TimeSpanElements
+        closed_time_spans[resource.id] = [
+            TimeSpanElement(
+                start_datetime=local_start_of_day(datetime.datetime.strptime(period["start_date"], "%Y-%m-%d").date()),
+                end_datetime=local_end_of_day(datetime.datetime.strptime(period["end_date"], "%Y-%m-%d").date()),
+                is_reservable=False,
+            )
+            for period in date_periods
+            # Overriding closed date periods are exceptions to the normal opening hours
+            if period["override"] and period["resource_state"] == HaukiResourceState.CLOSED.value
```

Comment:
Could extend the comment to clarify that `period["resource_state"]` is defined only for overrides in which case the state applies across all weekdays and hours (and `period["time_span_groups"]` are empty).

Reply:
Added better documentation for these directly to the `HaukiAPIDatePeriod` class